rtl/eclrtl/eclhelper_dyn.cpp

```diff
@@ -0,0 +1,278 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "platform.h"
+#include "jptree.hpp"
+#include "eclrtl.hpp"
+#include "eclhelper.hpp"
+#include "rtlds_imp.hpp"
+#include "eclhelper_base.hpp"
+#include "eclhelper_dyn.hpp"
+
+#include "rtlfield.hpp"
+#include "rtlrecord.hpp"
+#include "rtldynfield.hpp"
+#include "rtlkey.hpp"
+
+//---------------------------------------------------------------------------
+static const RtlRecordTypeInfo &loadTypeInfo(IPropertyTree &xgmml, IRtlFieldTypeDeserializer *deserializer, const char *key)
+{
+    StringBuffer xpath;
+    MemoryBuffer binInfo;
+    xgmml.getPropBin(xpath.setf("att[@name='%s_binary']/value", key), binInfo);
+    assertex(binInfo.length());
+    const RtlTypeInfo *typeInfo = deserializer->deserialize(binInfo);
+    assertex(typeInfo && typeInfo->getType()==type_record);
+    return *(RtlRecordTypeInfo *) typeInfo;
+}
+
+static void readString(StringBuffer &out, const char * &in)
+{
+    for (;;)
+    {
+        char c = *in++;
+        if (!c)
+            throw MakeStringException(0, "Invalid filter - missing closing '");
+        if (c=='\'')
+            break;
+        if (c=='\\')
+            UNIMPLEMENTED;
```

Comment:
Should be implemented (if only to read the next char and check it isn't null)

Reply:
It should - this bit is still a work in progress